src/app/main.cpp

```diff
@@ -149,6 +149,29 @@ static void gamescope_frametime(uint64_t app_frametime_ns, uint64_t latency_ns){
         HUDElements.gamescope_debug_latency.erase(HUDElements.gamescope_debug_latency.begin());
 }
 
+static EngineTypes engine_type_from_str(std::string engineName) {
```

Comment:
Maybe i am wrong, but SSO by default caps at 16 characters. So should not allocate under normal circumstances. Non the less, we actually don't need a full `std::string` either.
```suggestion
static EngineTypes engine_type_from_str(std::string_view engineName) {
```